lib/Driver/ToolChains/Flang.cpp

```diff
@@ -792,6 +795,22 @@ void FlangFrontend::ConstructJob(Compilation &C, const JobAction &JA,
   UpperCmdArgs.push_back("-output");
   UpperCmdArgs.push_back(ILMFile);
 
+  if(Args.getAllArgValues(options::OPT_fopenmp_targets_EQ).size() > 0) {
+    SmallString<128> TargetInfo;
+    Path = llvm::sys::path::parent_path(Output.getFilename());
+    Arg* Tgts = Args.getLastArg(options::OPT_fopenmp_targets_EQ);
+    assert(Tgts && Tgts->getNumValues() &&
+           "OpenMP offloading has to have targets specified.");
+    for (unsigned i = 0; i < Tgts->getNumValues(); ++i) {
+      if (i)
+        TargetInfo += ',';
+      llvm::Triple T(Tgts->getValue(i));
+      TargetInfo += T.getTriple();
+    }
```

Comment:
Line 799 through here might be rolled up into a little function that returns TargetInfo.str().

Reply:
Eric, I am not sure I understood this part correctly.

Reply:
Something like this
``` 
const char *
getOpenMPTargertInfoString() {
  SmallString<128> TargetInfo;
  Arg* Tgts = Args.getLastArg(options::OPT_fopenmp_targets_EQ);
  assert(Tgts && Tgts->getNumValues() &&
         "OpenMP offloading has to have targets specified.");
  for (unsigned i = 0; i < Tgts->getNumValues(); ++i) {
    if (i)
      TargetInfo += ',';
    llvm::Triple T(Tgts->getValue(i));
    TargetInfo += T.getTriple();
  }
  return TargetInfo.str();
}
.
.
.
UpperCmdArgs.push_back("-fopenmp-targets");
UpperCmdArgs.push_back(Args.MakeArgString(getOpenMPTargertInfoString()));
```